clients/src/main/java/org/apache/kafka/common/memory/MemoryPool.java

```diff
@@ -0,0 +1,106 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.memory;
+
+import org.apache.kafka.common.metrics.Sensor;
+
+import java.nio.ByteBuffer;
+
+
+/**
+ * A common memory pool interface for non-blocking pools.
+ * Every buffer returned from tryAllocate() must always be release()ed.
+ */
+public interface MemoryPool {
```

Comment:
I think it would be nice to avoid unnecessary naming inconsistencies between this and `BufferPool`. It may make sense to deviate in some cases, but could you take a pass and see if some names here or there should be renamed for consistency?